### PR TITLE
introduce gradle-util module for classpath isolation

### PR DESCRIPTION
**Pull request by neetopia**

### CODE REVIEW

File: gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -28,10 +28,11 @@ dependencies {
     // replace AGP dependency w/ gradle-api when we have source registering API available.
     compileOnly("com.android.tools.build:gradle:$agpBaseVersion")
     compileOnly(gradleApi())
-    implementation(project(":kotlin-analysis-api")) {
+    compileOnly(project(":kotlin-analysis-api")) {
         isTransitive = false
     }
-    implementation(project(":api"))
+    api(project(":api"))
```
Comment (ting-yuan) [inline gradle-plugin/build.gradle.kts:34]: compileOnly

File: gradle-plugin/build.gradle.kts [hunk 2]
```diff
@@ -28,10 +28,11 @@ dependencies {
     // replace AGP dependency w/ gradle-api when we have source registering API available.
     compileOnly("com.android.tools.build:gradle:$agpBaseVersion")
     compileOnly(gradleApi())
-    implementation(project(":kotlin-analysis-api")) {
+    compileOnly(project(":kotlin-analysis-api")) {
         isTransitive = false
```
Comment (ting-yuan) [inline gradle-plugin/build.gradle.kts:32]: remove isTransitive

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspAATask.kt [hunk 3]
```diff
@@ -58,7 +58,7 @@ abstract class KspAATask @Inject constructor(
         //  1. passing arguments by data structures in stdlib, or
         //  2. hoisting and publishing KspGradleConfig into another package.
 
-        val workerQueue = workerExecutor.noIsolation()
+        val workerQueue = workerExecutor.classLoaderIsolation()
```
Comment (ting-yuan) [inline gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspAATask.kt:61]: Isn't the purpose of the extra dependency removing classLoaderIsolation?

**Review by ting-yuan — COMMENTED** [review]
Can we name the module / artifact / package `common` or `common-deps` or etc, since it is not gradle specific and will be needed in command line and other build systems.

File: gradle-plugin/build.gradle.kts (hunk 1 above)
Comment (neetopia) [reply gradle-plugin/build.gradle.kts]: KSPLogger is still needed at runtime classpath, actually `implementation` makes more sense here.

**Review by ting-yuan — APPROVED** [review]

**Pull request merged**